inst/rmarkdown/templates/ajs_outbreak/skeleton/skeleton.Rmd

```diff
@@ -78,367 +78,341 @@ for (pkg in required_packages) {
-# linelist_dict$data_element_shortname
+## Stata data file -------------------------------------------------------------
+# linelist_raw <- rio::import(here("linelist.dat"))
+```
 
 
 
+<!-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+prep_nonDHIS_data
 
-## Read data -------------------------------------------------------------------
-## When you copy and paste file paths, you need to change \ to /.
-##
+This section is for data not from DHIS2. 
+If you have already loaded data from DHIS2, go to read_population_data.
 
-# Excel file ------------------------------------
-# to read in a specific sheet use "which"
-# linelist_raw <- rio::import(here("linelist.xlsx"), which = "Sheet1")
+It is more difficult to prepare the nonDHIS data. You can do it! It will just
+take a little more work.
 
+Checklist to update this script to match your data:
+[ ] Comment out all lines in read_DHIS_data and prep_DHIS_data
+[ ] Recode your variable names to match the dictionary
+[ ] Recode variable contents to match the dictionary
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
 
-# Excel file -- Specific range -------------------
-# You can specify a range in an excel sheet where your data are.
-# linelist_raw <- rio::import(here("linelist.xlsx"), range = "B2:J102")
+```{r prep_nonDHIS_data, warning = FALSE, message = FALSE}
+## MSF AJS Dictionary ----------------------------------------------------------
+## This step shows you the data dictionary.
 
+## The data dictionary has variable names in the data_element_shortname column. 
+## Possible values for each variable are specified in code and name columns.
+## Code has the shortened value and Name has the full-text value.
 
-# Excel file -- password protected ----------------
-# Use the excel.link package to load the file.
-#   The askpass package will prompt you to enter the password every time.
-# install.packages(c("excel.link", "askpass"))
-# library(excel.link)
-# 
-# linelist_raw <- xl.read.file(here("linelist.xlsx"),
-#                              xl.sheet = "Sheet1",
-#                              password = askpass::askpass(prompt = "please enter file password"))
+# get MSF standard dictionary for AJS 
+linelist_dict <- msf_dict("AJS", compact = FALSE) %>%
+  select(option_code, option_name, everything())
 
+## look at the standard dictionary by uncommenting the line below
+# View(linelist_dict) 
 
-# CSV file ----------------------------------------
-# linelist_raw <- rio::import(here("linelist.csv"))
+## You will need to recode your variables to match the data dictionary.
 
 
-# Stata data file ---------------------------------
-# linelist_raw <- rio::import(here("linelist.dat"))
 
+## Clean column names ----------------------------------------------------------
+## This step fixes the column names so they are easy to use in R.
 
+# make a copy of your orginal dataset and name it linelist_cleaned
+linelist_cleaned <- linelist_raw
```

Comment:
This is why the age pyramid is not working - because you are overwriting linelist_cleaned! 
I have set this chunk to eval = FALSE in latest commit